### PR TITLE
fix golangci-lint and yamllint errors

### PR DESCRIPTION
**Pull request by mfrancisc**
Fix golangci-lint and yamllint issues

### CODE REVIEW

**Review by xcoulon — APPROVED** [review]
Thanks @mfrancisc 🙌

**Comment by mfrancisc** [comment]
Not sure why few e2e test have failed, I've tried looking into the logs but didn't get the reason :( 

Only thing I can think I've changed that may have an impact are the yamllint issues, I can undo those if this can be the cause. Unless it's a random issue.

Anyone can help?

**Comment by xcoulon** [comment]
> Not sure why few e2e test have failed, I've tried looking into the logs but didn't get the reason :(
> 
> Only thing I can think I've changed that may have an impact are the yamllint issues, I can undo those if this can be the cause. Unless it's a random issue.
> 
> Anyone can help?

the error in the logs is 
```
Get "https://host-operator-metrics-service-toolchain-host-23075717.apps.ci-op-tm5plhn2-1de72.origin-ci-int-aws.dev.rhcloud.com/metrics": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```
this looks like an infra issue and most probably unrelated to your changes, so let's restart the e2e tests and see how it goes.

**Comment by mfrancisc** [comment]
> 

Thank you! I was very unlucky to hit this issue immediately :(

File: config/crd/bases/toolchain.dev.openshift.com_bannedusers.yaml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (ranakan19) [inline config/crd/bases/toolchain.dev.openshift.com_bannedusers.yaml:1]: is this change intentional or arising from a formatting difference? many other files also have similar changes (removing a blank line), so just wanted to confirm

Comment (mfrancisc) [reply config/crd/bases/toolchain.dev.openshift.com_bannedusers.yaml:1]: yes, yamllint was complaining about too many empty lines

**Review by ranakan19 — APPROVED** [review]
LGTM
